### PR TITLE
Fileglob Fix

### PR DESCRIPTION
**Pull request by sagagliardo**
Per https://github.com/ansible/ansible/issues/17136 , file glob has known issues in Ansible 2.2.0+ at the moment. Until this is resolved, I'm suggesting this PR to resolve this issue as an alternative to file glob.

Tests can be specified as either a single file or a path as an element in the tests array as well, as was previous behavior. The only change is that tests are now specified as an array vs a single string.

### CODE REVIEW

**Comment by sagagliardo** [comment]
Python 2.6 has some bugs with the functionality of lists and Ansible for some reason. When I bumped up Travis to 2.7 this was no longer an issue.

**Comment by sagagliardo** [comment]
@naftulikay Anything else I need to address here before we get this bad boy merged?

File: .travis.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@ language: python
 # support minimum python versions specified here:
 # https://docs.ansible.com/ansible/dev_guide/developing_modules_python3.html
 python:
-  - "2.6"
-  - "3.5"
+  - 2.7
+  - 3.5
```
Comment (naftulikay) [inline .travis.yml]: It seems that something automated mangled the Travis YAML file, let's undo these unnecessary changes. I specifically do this so that weird floating-point errors don't occur, YAML has a lot more types and a lot more implicit conversions than, say JSON.

**Comment by naftulikay** [comment]
I'm not entirely certain that this is even needed. Can you detail an exact use-case which this solves for?

File: .travis.yml (hunk 1 above)
Comment (sagagliardo) [reply .travis.yml]: Sure that's fine. If it was on the YAML file end we can try without.

**Comment by sagagliardo** [comment]
Looks like whatever it was being blocked by that was 2.7 specific is resolved, possibly by changes on your end? Passing now.

**Comment by sagagliardo** [comment]
@naftulikay any update getting this merged?

**Comment by naftulikay** [comment]
> I'm not entirely certain that this is even needed. Can you detail an exact use-case which this solves for?

If this is a bug in Ansible, then I don't see an urgency in fixing it in all downstream roles.

**Comment by naftulikay** [comment]
@sagagliardo Now that the refactor is complete, I do not see the need for this PR anymore, please see latest master and #6. 

There are three variables now that control file and directory copying:

```yaml
goss_file: goss.yml
goss_addtl_files: []
goss_addtl_dirs: []
```

These are their defaults. This is how they are used:

```yaml
- name: deploy test files
  copy: src={{ item }} dest={{ degoss_test_root }}
  with_items: "{{ [goss_file] + goss_addtl_files + goss_addtl_dirs }}"
  changed_when: degoss_changed_when
```
